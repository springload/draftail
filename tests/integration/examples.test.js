import path from 'path';
import fs from 'fs';
import { toHaveNoViolations } from 'jest-axe';
import { toMatchImageSnapshot } from 'jest-image-snapshot';

expect.extend(toHaveNoViolations);
expect.extend({ toMatchImageSnapshot });

const normalizePath = path.join(__dirname, 'normalize-rendering.css');
const normalizeRendering = fs.readFileSync(normalizePath, 'utf-8');

describe('/examples/', () => {
    let page;
    beforeAll(async () => {
        page = await global.BROWSER.newPage();
        await page.goto(`${global.ROOT}examples/`);

        await page.addScriptTag({ path: require.resolve('axe-core') });
    });

    it('loads', async () => {
        const text = await page.evaluate(() => document.body.textContent);
        expect(text).toContain('Write hereâ€¦');
    });

    it('axe', async () => {
        const text = await page.evaluate(() => {
            return new Promise((resolve, reject) => {
                window.axe.run((err, results) => {
                    if (err) reject(err);
                    resolve(results);
                });
            });
        });
        expect(text).toHaveNoViolations();
    });

    describe('simple editor', () => {
        it('renders', async () => {
            await page.addStyleTag({ content: normalizeRendering });

            await page.type('[data-mount] [role="textbox"]', 'Hello');
            await page.keyboard.press('Enter');
            await page.type('[data-mount] [role="textbox"]', '- ');
            await page.type('[data-mount] [role="textbox"]', 'World');

            const clip = await page.evaluate(() =>
                document
                    .querySelector('[data-mount]')
                    .getBoundingClientRect()
                    .toJSON(),
            );

            expect(await page.screenshot({ clip })).toMatchImageSnapshot({
                failureThresholdType: 'percent',
                // 0.2% difference to account for font rendering.
                failureThreshold: '0.002',
            });
        });
    });
});
