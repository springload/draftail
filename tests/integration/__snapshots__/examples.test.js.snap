// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/examples/ Wagtail features BOLD 1`] = `
Array [
  Object {
    "data": Object {},
    "depth": 0,
    "entityRanges": Array [],
    "inlineStyleRanges": Array [
      Object {
        "length": 4,
        "offset": 0,
        "style": "BOLD",
      },
    ],
    "text": "Bold",
    "type": "unstyled",
  },
]
`;

exports[`/examples/ Wagtail features BR 1`] = `
Object {
  "blocks": Array [
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "text": "Hel
lo",
      "type": "unstyled",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`/examples/ Wagtail features HORIZONTAL_RULE 1`] = `
Object {
  "blocks": Array [
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [
        Object {
          "key": 0,
          "length": 1,
          "offset": 0,
        },
      ],
      "inlineStyleRanges": Array [],
      "text": " ",
      "type": "atomic",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "text": "",
      "type": "unstyled",
    },
  ],
  "entityMap": Object {
    "0": Object {
      "data": Object {},
      "mutability": "IMMUTABLE",
      "type": "HORIZONTAL_RULE",
    },
  },
}
`;

exports[`/examples/ Wagtail features IMAGE 1`] = `
Object {
  "blocks": Array [
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "text": "",
      "type": "unstyled",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [
        Object {
          "key": 0,
          "length": 1,
          "offset": 0,
        },
      ],
      "inlineStyleRanges": Array [],
      "text": " ",
      "type": "atomic",
    },
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [],
      "inlineStyleRanges": Array [],
      "text": "",
      "type": "unstyled",
    },
  ],
  "entityMap": Object {
    "0": Object {
      "data": Object {
        "alt": "",
        "src": "../static/example-lowres-image.jpg",
      },
      "mutability": "MUTABLE",
      "type": "IMAGE",
    },
  },
}
`;

exports[`/examples/ Wagtail features LINK 1`] = `
Object {
  "blocks": Array [
    Object {
      "data": Object {},
      "depth": 0,
      "entityRanges": Array [
        Object {
          "key": 0,
          "length": 4,
          "offset": 0,
        },
      ],
      "inlineStyleRanges": Array [],
      "text": "Link",
      "type": "unstyled",
    },
  ],
  "entityMap": Object {
    "0": Object {
      "data": Object {
        "url": "http://www.example.com/",
      },
      "mutability": "MUTABLE",
      "type": "LINK",
    },
  },
}
`;

exports[`/examples/ Wagtail features header-three 1`] = `
Array [
  Object {
    "data": Object {},
    "depth": 0,
    "entityRanges": Array [],
    "inlineStyleRanges": Array [],
    "text": "H3",
    "type": "header-three",
  },
]
`;

exports[`/examples/ Wagtail features unordered-list-item 1`] = `
Array [
  Object {
    "data": Object {},
    "depth": 0,
    "entityRanges": Array [],
    "inlineStyleRanges": Array [],
    "text": "UL",
    "type": "unordered-list-item",
  },
]
`;
